autrainer/transforms/smart_compose.py

```diff
@@ -72,13 +72,16 @@ def _sort(self):
         """Sort the transforms by their order attribute if present."""
         self.transforms.sort(key=lambda x: getattr(x, "order", 0))
 
-    def get_collate_fn(self, data: "AbstractDataset") -> Optional[Callable]:
+    def get_collate_fn(
+        self, data: "AbstractDataset", default: Callable
```

Comment:
Why is it beneficial to have the default collate function as argument here (or not provide the default directly)?
The intention was that augmentations could specify [`get_collate_fn`](https://autrainer.github.io/autrainer/modules/augmentations.html#augmentations-with-collate-functions) and do essentially whatever they want with the data. In some cases it makes sense to just use `torch.utils.data.default_collate`, however we should then just document that we have a custom one instead of providing it always?